Libraries/LibWeb/MediaCapabilitiesAPI/MediaCapabilities.cpp

```diff
@@ -29,4 +127,118 @@ void MediaCapabilities::initialize(JS::Realm& realm)
+MediaCapabilitiesDecodingInfo MediaCapabilities::create_a_media_capabilities_decoding_info(MediaDecodingConfiguration configuration) const
+{
+    // 1. Let info be a new MediaCapabilitiesDecodingInfo instance. Unless stated otherwise, reading and
+    //    writing apply to info for the next steps.
+    MediaCapabilitiesDecodingInfo info = {};
+
+    // 2. Set configuration to be a new MediaDecodingConfiguration. For every property in configuration create
+    //    a new property with the same name and value in configuration.
+    info.configuration = { { configuration.video, configuration.audio }, configuration.type, configuration.key_system_configuration };
+
+    // 3. If configuration.keySystemConfiguration exists:
+    if (false) {
+        // FIXME: Implement this.
+    }
+    // 4. Otherwise, run the following steps:
+    else {
+        // 1. Set keySystemAccess to null.
+        // FIXME: Implement this.
+
+        // 2. If the user agent is able to decode the media represented by configuration, set supported to true.
+        // 3. Otherwise, set it to false.
+        info.supported = is_able_to_decode_media(configuration);
+    }
+
+    // 5. If the user agent is able to decode the media represented by configuration at the indicated framerate without
+    //    dropping frames, set smooth to true. Otherwise set it to false.
+    // FIXME: Actually check this.
+    info.smooth = false;
+
+    // 6. If the user agent is able to decode the media represented by configuration in a power efficient manner, set
+    //    powerEfficient to true. Otherwise set it to false.
+    // FIXME: Actually check this... somehow.
+    info.power_efficient = false;
+
+    // 7. Return info.
+    return info;
+}
+
+bool MediaCapabilities::is_able_to_decode_media(MediaDecodingConfiguration configuration) const
+{
+    if (configuration.type != Bindings::MediaDecodingType::MediaSource)
+        return false;
+
+    if (configuration.video.has_value()) {
+        auto video_mime_type = MimeSniff::MimeType::parse(configuration.video.value().content_type);
+        if (!video_mime_type.has_value() || !Web::HTML::HTMLMediaElement::supported_video_subtypes.contains_slow(video_mime_type->subtype()))
+            return false;
+    }
+
+    if (configuration.audio.has_value()) {
+        auto audio_mime_type = MimeSniff::MimeType::parse(configuration.audio.value().content_type);
+        if (!audio_mime_type.has_value() || !Web::HTML::HTMLMediaElement::supported_audio_subtypes.contains_slow(audio_mime_type->subtype()))
+            return false;
+    }
+
+    return true;
+}
+
+WebIDL::ExceptionOr<GC::Ref<JS::Object>> MediaCapabilitiesDecodingInfo::to_object(JS::Realm& realm)
+{
+    auto object = JS::Object::create(realm, realm.intrinsics().object_prototype());
+
+    // FIXME: Also include configuration in this object.
+
+    TRY(object->create_data_property("supported", JS::BooleanObject::create(realm, supported)));
```

Comment:
I believe all of these can be MUST'd, which would allow us to remove the ExceptionOr (which the caller ends up verifying succeeds anyhow)

Reply:
Done.